js/utils.js

```diff
@@ -0,0 +1,65 @@
+async function makeApiCall(
+  url,
+  method = 'get',
+  body = null,
+  credentials = 'include',
+  headers = { 'content-type': 'application/json' },
+  options = null,
+) {
+  try {
+    const response = await fetch(url, {
+      method,
+      body,
+      headers,
+      credentials,
+      ...options,
+    });
+    const data = await response.json();
+    const status = response.status;
+    const res = {
+      data,
+      status,
+    };
+    return res;
+  } catch (err) {
+    throw err;
+  }
+}
+
+function createElement({ type, classList = [], id }) {
+  const element = document.createElement(type);
+  element.classList.add(...classList);
+  if (!id) {
+    element.id = isEmpty(id);
+  } else {
+    element.id = id;
+  }
+  return element;
+}
+
+/**
+ * @param value:
+ * @returns boolean which returns
+ * - `true` if value is empty or falsy
+ * - `false`  if value is not empty or truthy
+ */
+function isEmpty(valueToCheck) {
```

Comment:
Do we really need such a verbose function?

Reply:
https://github.com/Real-Dev-Squad/website-www/pull/432#discussion_r1153364091